aiosmtpd/tests/test_server.py

```diff
@@ -5,18 +5,39 @@
 
 import platform
 import socket
+import ssl
+import time
+from contextlib import ExitStack
 from functools import partial
+from pathlib import Path
 
 import pytest
 from pytest_mock import MockFixture
 
-from aiosmtpd.controller import Controller, _FakeServer
+from aiosmtpd.controller import Controller, _FakeServer, UnixSocketController
 from aiosmtpd.handlers import Sink
 from aiosmtpd.smtp import SMTP as Server
 
 from .conftest import Global
 
 
+class SlowStartController(Controller):
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault("ready_timeout", 0.5)
+        super().__init__(*args, **kwargs)
+
+    def _run(self, ready_event):
+        time.sleep(self.ready_timeout * 1.1)
+        try:
+            super()._run(ready_event)
+        except Exception:
+            pass
```

Comment:
Now that we actually react to timeouts in `Controller`, we need to simulate a timeout.